app/models/spree/order.rb

```diff
@@ -615,6 +616,20 @@ def sorted_line_items
       end
     end
 
+    def can_generate_new_invoice?
+      return true if invoices.empty?
+
+      invoices.last.data["order"].to_json != to_json(include: :line_items)
+    end
+
+    def generate_invoice(number = invoices.size + 1, date = Time.now)
+      invoices.create({
+                        number: number,
+                        date: date,
+                        data: to_json(include: :line_items)
```

Comment:
Is `to_json` already creating a string? I thought the model was expecting a hash and it's serialized when stored to the database.